### ISSUE TITLE
Missing some interact methods in ElementBuilder

### ISSUE THREAD

**Issue by spaceemotion** [nifty-gui/nifty-gui#290]
Hey, I just noticed that you can set some "default" interact parameters via code, but not all of them. It seems that one cannot access the interact attributes either. Am I missing something? I really need to define different behaviours for left, right and middle clicks and need to do that via code.

Thanks in advance!

**Comment by relu91** [comment]
Could you please be more precise? What are you trying to do? Are you considering to use event bus to handle interact events?

**Comment by spaceemotion** [comment]
No, I am using the interact stuff as explained on page 69 in the manual (yes, I am actually reading that ^^). It is using the `<interact />` tag in XML form, but in java this is very limited as it only allows the basic parameters like `onClick` and `onRelease`, but not `onPrimaryClick`.

I currently found a way to circumvent this by extending the ControlDefinitionBuilder and accessing the field through a protected function (yep, the only way…). I wish there would be an easier way to do this, like with the `attributes.set(key, value)`.

**Comment by spaceemotion** [comment]
@relu91 Any idea when you have some free time to add this?

**Comment by relu91** [comment]
Hey sorry for the delay, I think I'll add this after I've fished my last pull request! :)

**Comment by bgroenks96** [comment]
@relu91 Is this addressed in your current pending pull request? If so, could you add a 'closes' #290 in its description?

**Comment by ghost** [comment]
@bgroenks96 No, this has nothing to do with #288.

**Comment by relu91** [comment]
Yup, it's not link in anyway with #288 I just said that I will work on that after this pull request will be merged (due to lack of time). @bgroenks96 you can work on this issue if you want!

**Comment by bgroenks96** [comment]
Ok I misunderstood. Whoever has time first then @relu91 :smile:

**Comment by bgroenks96** [comment]
This issue has the ready label, is there a fix pending somewhere? @relu91 @void256

**Comment by void256** [comment]
I'm not working on it.

**Comment by relu91** [comment]
Me neither, sorry this was a little change but now I can't figure out where should I look and I've no time to search in the code... :cry:

**Comment by bgroenks96** [comment]
I guess I would start with the ElementBuilder class? :laughing: 

But in all seriousness... I don't really know. I'm not even entirely clear on what needs to be changed for this issue. I was just going through nifty issues trying to make sure as many things as possible are being addressed in some way.
